components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/add-security-questions.jsp

```diff
@@ -0,0 +1,145 @@
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+<%@ page contentType="text/html;charset=UTF-8" pageEncoding="UTF-8" %>
+
+<%@ page import="org.owasp.encoder.Encode" %>
+
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.AuthenticationEndpointUtil" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.EncodedControl" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
+<%@ page import="org.wso2.carbon.identity.recovery.ui.IdentityManagementAdminClient" %>
+<%@ page import="org.wso2.carbon.identity.recovery.model.ChallengeQuestion" %>
+<%@ page import="org.wso2.carbon.utils.ServerConstants" %>
+<%@ page import="org.wso2.carbon.ui.CarbonUIUtil" %>
+<%@ page import="org.wso2.carbon.CarbonConstants" %>
+
+<%@ page import="java.nio.charset.StandardCharsets" %>
+<%@ page import="java.util.ResourceBundle" %>
+<%@ page import="java.util.ArrayList" %>
+<%@ page import="java.util.Iterator" %>
+<%@ page import="java.util.HashMap" %>
+<%@ page import="java.util.List" %>
+<%@ page import="java.util.Map" %>
+
+<%
+    String BUNDLE = "org.wso2.carbon.identity.application.authentication.endpoint.i18n.Resources";
+    ResourceBundle resourceBundle = ResourceBundle.getBundle(BUNDLE, request.getLocale(), new
+            EncodedControl(StandardCharsets.UTF_8.toString()));
+    String urlData = request.getParameter("data");
+    // Extract the challenge questions from the request and add them into an array
+    String[] questionSets = null;
+    if (urlData != null) {
+        questionSets = urlData.split("&");
+    }
+    // Hash-map to hold available challenge questions in the system
+    Map<String, List<ChallengeQuestion>> challengeQuestionMap = new HashMap<>();
+
+    for (String question : questionSets) {
+        String[] questionProperties = question.split("\\|");
+        // Construct a new ChallengeQuestion for each challenge question received from the request
+        ChallengeQuestion tempChallengeQuestion = new ChallengeQuestion();
+        // Extract the challenge question properties
+        String questionSetId = questionProperties[0];
+        String questionId = questionProperties[1];
+        String questionBody = questionProperties[2];
+        tempChallengeQuestion.setQuestionSetId(questionSetId);
+        tempChallengeQuestion.setQuestionId(questionId);
+        tempChallengeQuestion.setQuestion(questionBody);
+        // Add the challenge question to the Hash-map
+        List<ChallengeQuestion> challengeQuestionList = challengeQuestionMap.get(questionProperties[0]);
```

Comment:
You can reuse the local variable questionSetId instead of getting the value again from array.

Reply:
fixed